collect_app/src/test/java/org/odk/collect/android/application/initialization/ExistingSettingsMigratorTest.kt

```diff
@@ -0,0 +1,49 @@
+package org.odk.collect.android.application.initialization
+
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.Matchers.`is`
+import org.hamcrest.Matchers.nullValue
+import org.junit.Test
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+import org.odk.collect.android.preferences.source.SettingsProvider
+import org.odk.collect.projects.InMemProjectsRepository
+import org.odk.collect.projects.Project
+import org.odk.collect.testshared.InMemSettings
+
+class ExistingSettingsMigratorTest {
+
+    @Test
+    fun `migrates general and admin settings for each project`() {
+        val projectsRepository = InMemProjectsRepository()
+        val project1 = projectsRepository.save(Project.New("1", "1", "#ffffff"))
+        val project2 = projectsRepository.save(Project.New("2", "2", "#ffffff"))
+
+        val project1GeneralSettings = InMemSettings()
+        val project1AdminSettings = InMemSettings()
+        val project2GeneralSettings = InMemSettings()
+        val project2AdminSettings = InMemSettings()
+
+        val settingsProvider = mock<SettingsProvider> {
```

Comment:
I think it'd be nice to have an `InMemSettingsProvider` that does the work for me here, as the test setup is pretty verbose. There are a few other tests that we could clean up with that as well, so I'll follow up with a PR for that rather than holding this up.